soa/service/xml_helpers.h

```diff
@@ -90,5 +91,15 @@ T extractDef(const tinyxml2::XMLDocument & doc,
     return extractDef<T>(element, path, def);
 }
 
+inline std::string
+xmlDocumentAsString(const tinyxml2::XMLDocument & xmlDocument)
```

Comment:
That's because it is a good practice to constify refs by default and that XMLDocument::Print is not declared as const-safe even though it is.